lib/cisco_node_utils/snmp_notification_receiver.rb

```diff
@@ -0,0 +1,158 @@
+# November 2015, Jonathan Tripathy
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+
+module Cisco
+  # SnmpNotificationReceiver - node utility class for SNMP server management
+  class SnmpNotificationReceiver < NodeUtil
+    attr_reader :name
+
+    def initialize(name,
+                   instantiate=true,
+                   type='',
+                   version='',
+                   security='',
+                   username='',
+                   port='',
+                   vrf='',
+                   source_interface='')
```

Comment:
Consider using named args instead:

```ruby
def initialize(name,
               instantiate: true, type: '', version: '', security: '', username: '',
               port: '', vrf: '', source_interface: '')
```

This will make the instantiation calls easier to read and less error prone:

```ruby
    receiver = Cisco::SnmpNotificationReceiver.new(
      id,
      instantiate:      true,
      type:             'informs',
      version:          '3',
      security:         'priv',
      username:         'ab',
      port:             '45',
      vrf:              'red',
      source_interface: 'ethernet1/3')
```